### DISCUSSION TITLE
Attempt at XUnit unit testing leads to WinRT.ActivationFactory errors

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [microsoft/WindowsAppSDK#2696] [Q&A]
I've been using XUnit to develop unit tests for .NET 6 class libraries.  Now I want to do unit testing for Windows APP SDK class libraries.  Is it even possible to unit test a WinUI 3 class library? 

I modified a .NET 6 XUnit project to target net6.0-windows10.0.19041.0.  This enables the Test Explorer to run the XUnit test and do simple .NET 6 assertions about native .NET types.  However, the moment I try and reference something in Windows App SDK, for example Microsoft.UI.Text.FontWeights, I get WinRT.ActivationFactory errors.  How can I overcome this?
![WinRT Factory Error](https://user-images.githubusercontent.com/78389339/177014302-9c4e3531-cf17-45e7-bbc3-308904d44ef5.jpg)

**Comment by DrusTheAxe** [comment]
How is your XUnit test built?

I assume your project is just producing a DLL i.e. it's unpackaged, not [MSIX] packaged.

Is it using the Bootstrapper?

Does your project specify `<WindowsAppSDKSelfContained>true</>`?
